freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/instance/azure/AzureInstanceTemplateParameters.java

```diff
@@ -0,0 +1,83 @@
+package com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.instance.azure;
+
+import java.util.Map;
+
+import javax.validation.Valid;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.sequenceiq.cloudbreak.cloud.model.instance.AzureInstanceTemplate;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.cloudbreak.common.mappable.MappableBase;
+import com.sequenceiq.common.api.type.EncryptionType;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.doc.FreeIpaModelDescriptions;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel(value = "AzureInstanceTemplateParameters")
+public class AzureInstanceTemplateParameters extends MappableBase {
```

Comment:
As mentioned below, I am not sure it is worth the effort to introduce all of this at FreeIPA and DL API level. Instead, like for DH creation, disk encryption details could be directly extracted from the `EnvironmentResponse` in the appropriate converter below the controller / service method.